packages/components/src/RBACProvider/RBACProvider.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useState } from 'react';
+import { Access } from '@redhat-cloud-services/rbac-client';
+import { Bullseye, Spinner } from '@patternfly/react-core';
+
+import {
+  RBACContext,
+  doesHavePermissions,
+  getRBAC,
+  hasAllPermissions,
+  initialPermissions,
+} from '@redhat-cloud-services/frontend-components-utilities/RBAC';
+
+const hasAccessWithUserPermissions = (userPermissions: Access[]) => {
+  return (requiredPermissions: any, checkAll = true) => {
+    return checkAll ? hasAllPermissions(userPermissions, requiredPermissions) : doesHavePermissions(userPermissions, requiredPermissions);
+  };
+};
+
+export interface RBACProviderProps {
+  appName: string;
+}
+
+export const RBACProvider: React.FunctionComponent<RBACProviderProps> = ({ appName, children }) => {
+  const [permissionState, setPermissionState] = useState(initialPermissions);
+  let loading = false;
```

Comment:
You probably want to store this value in a `useRef`. The value can be potentially re-declared during a render and you can be dealing with a race condition.

Reply:
Changed to using ref.